docs/api_reference.rst

```diff
@@ -22,24 +22,88 @@ Classes for creating Generalized Linear Models (GLMs) for both single neurons an
 The ``nemos.basis`` module
 --------------------------
 Provides basis function classes to construct and transform features for model inputs.
+Basis can be grouped according to the mode of operation into basis that performs convolution and basis that operates
+as non-linear maps.
 
-.. currentmodule:: nemos.basis
+
+**The Abstract Classes:**
+
+These classes are the building blocks for the concrete basis classes.
+
+.. currentmodule:: nemos.basis._basis
 
 .. autosummary::
-    :toctree: generated/basis
+    :toctree: generated/_basis
     :recursive:
     :nosignatures:
 
     Basis
+
+.. currentmodule:: nemos.basis._spline_basis
+.. autosummary::
+    :toctree: generated/_basis
+    :recursive:
+    :nosignatures:
+
     SplineBasis
-    BSplineBasis
-    CyclicBSplineBasis
-    MSplineBasis
-    OrthExponentialBasis
-    RaisedCosineBasisLinear
-    RaisedCosineBasisLog
+
+
+**Bases For Convolution:**
+
+.. currentmodule:: nemos.basis
+
+.. autosummary::
+    :toctree: generated/basis
+    :recursive:
+    :nosignatures:
+
+
+    MSplineConv
+    BSplineConv
+    CyclicBSplineConv
+    RaisedCosineLinearConv
+    RaisedCosineLogConv
+    OrthExponentialConv
+
+.. check for a config that prints only nemos.basis.Name
+
+**Bases For Non-Linear Mapping:**
+
+.. currentmodule:: nemos.basis
+
+.. autosummary::
+    :toctree: generated/basis
+    :recursive:
+    :nosignatures:
+
+    MSplineEval
+    BSplineEval
+    CyclicBSplineEval
+    RaisedCosineLinearEval
+    RaisedCosineLogEval
+    OrthExponentialEval
+
+**Composite Bases:**
+
+.. currentmodule:: nemos.basis._basis
+
+.. autosummary::
+    :toctree: generated/_basis
+    :recursive:
+    :nosignatures:
+
     AdditiveBasis
     MultiplicativeBasis
+
+**Basis As `scikit-learn` Tranformers:**
```

Comment:
I have a suggestion for future PRs, it is probably quicker if you push changes like this directly, committing one at the time on github is slow, since it always scrolls up.